sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUdfs.scala

```diff
@@ -179,7 +179,12 @@ private[hive] case class HiveGenericUdf(funcWrapper: HiveFunctionWrapper, childr
         })
       i += 1
     }
-    unwrap(function.evaluate(deferedObjects), returnInspector)
+    
+    if (function.getUdfName().endsWith("UDFCurrentDB")) {
```

Comment:
This seems a hack to me, can you create a UDF instead of intercepting code here?